bundles/framework/divmanazer/scss/badge.scss

```diff
@@ -21,16 +21,26 @@
 }
 
 .oskari-badge {
+    position: relative;
     font-weight: bold;
     color: #ffffff;
     vertical-align: baseline;
     white-space: nowrap;
     text-shadow: 0 -1px 0 rgba(0, 0, 0, 0.25);
     background-color: #999999;
-    padding: 1px 9px 2px;
+    padding: 1px 7px 2px;
+    top: 50%;
+    left: 50%;
+    transform: translate(0,-50%);
     @include border-radius(9px);
 }
-
+.badge-wrapper {
+    display: inline-block;
+    position: absolute;
+    right: 8%;
+    height: inherit;
+    max-width: 50px;
```

Comment:
I'd say "right: 8%" or "max-width" defs should NOT be part of the badge ui-component. It should be defined in for example layerselector bundles css and specific to the layerlisting UI.